cloud/amazon/cloudwatchevent_rule.py

```diff
@@ -0,0 +1,409 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: cloudwatchevent_rule
+short_description: Manage CloudWatch Event rules and targets
+description:
+  - This module creates and manages CloudWatch event rules and targets.
+version_added: "2.2"
+extends_documentation_fragment:
+  - aws
+author: "Jim Dalton (@jsdalton) <jim.dalton@gmail.com>"
+requirements:
+  - python >= 2.6
+  - boto3
+notes:
+  - A rule must contain at least an I(event_pattern) or I(schedule_expression). A
+    rule can have both an I(event_pattern) and a I(schedule_expression), in which
+    case the rule will trigger on matching events as well as on a schedule.
+  - When specifying targets, I(input) and I(input_path) are mutually-exclusive
+    and optional parameters.
+options:
+  name:
+    description:
+      - The name of the rule you are creating, updating or deleting. No spaces
+        or special characters allowed (i.e. must match C([\.\-_A-Za-z0-9]+))
+    required: true
+  schedule_expression:
+    description:
+      - A cron or rate expression that defines the schedule the rule will
+        trigger on. For example, C(cron(0 20 * * ? *)), C(rate(5 minutes))
+    required: false
+  event_pattern:
+    description:
+      - A string pattern (in valid JSON format) that is used to match against
+        incoming events to determine if the rule should be triggered
+    required: false
+  state:
+    description:
+      - Whether the rule is present (and enabled), disabled, or absent
+    choices: ["present", "disabled", "absent"]
```

Comment:
Typically when there's an enable/disable distinction, Ansible uses 4 states: 

- disabled (the rule must exist and be off)
- active (the rule must exist and be on)
- present (the rule must exist, but won't be turned on or off)
- absent (the rule must not exist)